packages/toolpad-app/src/components/EditableText.tsx

```diff
@@ -1,42 +1,38 @@
 import * as React from 'react';
-import {
-  TextField,
-  Theme,
-  TypographyVariant,
-  SxProps,
-  InputBaseComponentProps,
-} from '@mui/material';
+import { TextField, Theme, TypographyVariant, SxProps } from '@mui/material';
 
 interface EditableTextProps {
   defaultValue?: string;
+  disabled?: boolean;
   editing?: boolean;
```

Comment:
In terms of naming maybe it makes more sense to go use the adjective `editable` instead of the verb?

Reply:
Do you mean `editable` instead of `editing`? I feel like it conveys different things: `editable` would mean whether the component can ever be edited; whereas `editing` to me conveys the idea that this prop is tied to its "state"

Reply:
My initial thoughts were the same, however if we looked at https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Editable_content - it kinda works and does what we are trying to do here as well, no?
(it's not without a reason why naming in programming is one of the most difficult things :D )

Reply:
Do you mean using a `Typography` and toggle `contentEditable` instead of using a `TextField`? It's an interesting idea, but we'd have to disable pasting formatted text and deal with accessibility and so on. I don't think it will necessarily make things simpler.

Or do you just propose to name the prop `contentEditable`? I think the existence of the prop illustrates that there's a preference of "editable" over "editing". i don't necessarily think we should use `contentEditable`, we're not really making the "content" editable the same way `contentEditable` does.

Reply:
Just adding my opinion - I agree with Bharat in the sense that something like `editable: false` to me would mean the component would be read-only or something...

Then again if the `editing` prop is not internal state and comes from outside the component, maybe the prop should express something like an "edit mode"?

Reply:
> Do you mean using a Typography and toggle contentEditable instead of using a TextField? It's an interesting idea

Oh sorry, my bad, I just added more confusion 😅 

No, I just wanted to refer to the prop name that is used `contentEditable` in our case that could be `editable` just as you proposed. I was also on the verge of whether it should `editing` but my choice went with using same format as `contentEditable` uses 🙂

Reply:
I think `contentEditable` expresses a "permanent" nature of the element as being able to be edited; whereas what I was trying to express with the prop name is the "temporary" state of the component as either being able to be edited or not.

Reply:
I think `contentEditable` expresses a "permanent" nature of the element as being able to be edited, whereas I was trying to express that the prop controls only the "temporary" state of the component, which is why "editing".  

Perhaps "focussed" could be clearer?

Reply:
Overall I think that neither name would is wrong so choosing either should be fine. I'd say we should pick one which is preferred by majority (don't count me). After all, after such a long debate I don't think either of us will be confused by what `editing` or `editable` for this component means 😅 

Just my 2 last cents which might be inline to what Pedro wrote - to me `editing` may sound a little bit more like a state defining value that we pass around, where `editable` defines kinda of interface so in the end it could be even something like this:
```jsx
const [editing, setEditing] = useState()
...
return <Field editable={editing} />
```

Reply:
> ```js
> const [editing, setEditing] = useState()
> ...
> return <Field editable={editing} />
> ```

That makes sense to me. We can go ahead with `editable`